engine/runtime-parser/src/main/java/org/enso/compiler/core/EnsoParser.java

```diff
@@ -6,48 +6,23 @@
 import org.enso.compiler.core.ir.Location;
 import org.enso.compiler.core.ir.Module;
 import org.enso.syntax2.Parser;
-import org.enso.syntax2.Tree;
 
-public final class EnsoParser implements AutoCloseable {
-  private final Parser parser;
-
-  public EnsoParser() {
-    Parser p;
-    try {
-      p = Parser.create();
-    } catch (LinkageError err) {
-      err.printStackTrace();
-      throw err;
-    }
-    this.parser = p;
+public final class EnsoParser {
```

Comment:
Please put here `private EnsoParser() {}` so nobody can create instances of this class - without explicit `private`  constructor a default `public` constructor is generated. The _"beauty"_ of API design in OOP...